### DISCUSSION TITLE
Filesystem Full Behavior?

### DISCUSSION THREAD

**Discussion by AlaskanAstro** [aaronwmorris/indi-allsky#1394] [Q&A]
Hello, just trying to figure out what the default behavior is supposed to be when the SD card inevitably fills up. Does it automatically delete the oldest images and videos first? Do I need to manually clear things out? Is there an age setting I might have remembered seeing that I need to adjust? 
Thanks

**Comment by aaronwmorris** [comment]
There are two settings in the config under the Image tab for expiring images and timelapses.  The default is 30 days for images (and panoramas) and 365 days for timelapses (videos, keograms, startrails).

As of this point, I am going move these two settings from the Image tab to the Admin tab.  I think that makes more sense.

Also, I just now realized I only have the expiration firing during the transition from night to day.  This is obviously a problem if there is no day-night transition.  I am going to adjust this so that things are expired at every transition (day->night, night->day, night->night, and day->day).

**Comment by AlaskanAstro** [comment]
So if the system fills up before the date, does it just stop saving new images?

**Reply by aaronwmorris** [reply]
It will continue to try to save images, but part of the process will fail.  Once space reaches 90% on any of the filesystems, a notification will also be generated for the user on the web interface.

**Comment by AlaskanAstro** [comment]
Ok thanks. I'll need to tune that. 4k images and videos fill things up in a week or 2.

**Reply by aaronwmorris** [reply]
I have noticed that about the 4k systems I have tested.  Even with 256GB sdcards, it has a tendency to fill up very quickly.  5-10 days of images may be more appropriate in those cases.
